File: server/model/montage.go
package model

import "gorm.io/datatypes"

type MontageInput struct {
	Brief           string             `json:"brief,omitempty"`
	PipelineKey     string             `json:"pipeline_key,omitempty"`
	SourceAssets    []MontageAsset     `json:"source_assets,omitempty"`
	Preferences     MontagePreferences `json:"preferences,omitempty"`
	DeliveryTargets []string           `json:"delivery_targets,omitempty"`
	Advanced        map[string]any     `json:"advanced,omitempty"`
}

type MontageAsset struct {
	Type       string `json:"type"`
	URL        string `json:"url,omitempty"`
	TaskFileID string `json:"task_file_id,omitempty"`
	Text       string `json:"text,omitempty"`
	FileName   string `json:"file_name,omitempty"`
	MimeType   string `json:"mime_type,omitempty"`
	FileSize   int64  `json:"file_size,omitempty"`
}

type MontagePreferences struct {
	AspectRatio     string `json:"aspect_ratio,omitempty"`
	DurationSeconds int64  `json:"duration_seconds,omitempty"`
	Style           string `json:"style,omitempty"`
	MusicPrompt     string `json:"music_prompt,omitempty"`
	SubtitleMode    string `json:"subtitle_mode,omitempty"`
	VoiceoverMode   string `json:"voiceover_mode,omitempty"`
}

type MontageDefaults struct {
	DefaultPipeline string             `json:"default_pipeline,omitempty"`
	Preferences     MontagePreferences `json:"preferences,omitempty"`
	AssetGuidance   string             `json:"asset_guidance,omitempty"`
	DeliveryTargets []string           `json:"delivery_targets,omitempty"`
}

func (t *Task) SetMontageInput(input MontageInput) {
	t.MontageInput = datatypes.NewJSONType(input)
}

func (p *Plan) SetMontageInput(input MontageInput) {
	p.MontageInput = datatypes.NewJSONType(input)
}

func (p *Project) SetMontageDefaults(defaults MontageDefaults) {
	p.MontageDefaults = datatypes.NewJSONType(defaults)
}

File: studio/src/types/montage.ts
export type MontageAssetType = 'text' | 'image_url' | 'video' | 'video_url' | 'audio' | 'audio_url' | 'document_url'

export interface MontageAsset {
  type: MontageAssetType
  url?: string
  task_file_id?: string
  text?: string
  file_name?: string
  mime_type?: string
  file_size?: number
}

export interface MontagePreferences {
  duration_seconds?: number
  style?: string
  music_prompt?: string
  subtitle_mode?: string
  voiceover_mode?: string
}

export interface MontageInput {
  brief?: string
  pipeline_key?: string
  source_assets?: MontageAsset[]
  preferences?: MontagePreferences
  delivery_targets?: string[]
  advanced?: Record<string, unknown>
}

export type MontageSourceRequirement = 'optional' | 'video' | 'video_or_audio'
export type MontageOutputMode = 'single' | 'multiple'

export interface MontagePipelineCapability {
  key: string
  display_name: string
  description: string
  best_for: string[]
  source_hint: string
  output_hint: string
  source_requirement: MontageSourceRequirement
  output_mode: MontageOutputMode
  recommended_duration_seconds: number
}

export interface MontageCapabilityListResponse {
  enabled: boolean
  default_pipeline: string
  max_duration_seconds: number
  max_assets: number
  items: MontagePipelineCapability[]
}
